api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -105,6 +107,16 @@ public CohortDefinition getAnyEncounterDuringPeriod() {
         return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate");
     }
 
+    @DocumentedDefinition("codedObsSearchAdvanced")
+    public CohortDefinition getPatientsHavingObs() {
+        CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
```

Comment:
The idea was that "advanced" would support all the parameters (though we might add them as needed, but we don't want to keep having to change the name each time we add new parameters).

Definitely the name of this method is wrong, and it should match the library key. Also the name of the method below this one is wrong (getAnyEncounterDuringPeriodWithOccurrence).

I agree that as-written nothing about this is specific to coded obs, but I assume @andela-cnwankwo is planning to also add the values. Right?

Mike, I wonder if we should create a cohort definition that is smart enough to look a the concept's datatype and delegate to the right cohort definition. As things stand now I feel like the Andela team is going to have to create a different library definition for each datatype...